slack-notify.sh

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+if [[ -z "${SLACK_TOKEN-}" ]]; then
+    read -r SLACK_TOKEN
+fi
+SLACK_HOOK_URL="https://hooks.slack.com/services/${SLACK_TOKEN}"
+
+: "${GITHUB_SERVER_URL:=https://github.com}"
+: "${GITHUB_API_URL:=https://api.github.com}"
+: "${GITHUB_SHA:=$(git rev-parse HEAD)}"
+: "${PICTURE_BASE_URL:=https://github.com/foxgoat/howl/raw/master}"
+
+if [[ -z "${GITHUB_REPOSITORY-}" ]]; then
+    REPO_URL=$(git remote get-url origin)
+    GITHUB_REPOSITORY="${REPO_URL#${GITHUB_SERVER_URL}/}"
+else
+    REPO_URL="${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY}"
+fi
+REPO=${GITHUB_REPOSITORY#*/}
+
+: "${BUILD_URL:=${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY}/actions${GITHUB_RUN_ID:+/runs/${GITHUB_RUN_ID}}}"
+: "${GITHUB_REF:=$(git symbolic-ref HEAD)}"
+BRANCH=${GITHUB_REF#refs/heads/}
+BRANCH_URL="${REPO_URL}/tree/${BRANCH}"
+
+# prop extracts property from JSON object.
+# prop('{"name": "Tricia", "age": 42}', age)
```

Comment:
nit: this comment example is in a language that uses parens and commas to call functions. Bash uses spaces..

Reply:
so sorry.